compute_tools/src/spec.rs

```diff
@@ -397,10 +397,33 @@ pub fn handle_databases(spec: &ComputeSpec, client: &mut Client) -> Result<()> {
                 // We do not check either DB exists or not,
                 // Postgres will take care of it for us
                 "delete_db" => {
-                    let query: String = format!("DROP DATABASE IF EXISTS {}", &op.name.pg_quote());
+                    // In Postgres we can't drop a database if it is a template.
+                    // So we need to unset the template flag first, but it could
+                    // be a retry, so we could've already dropped the database.
+                    // Check that database exists first to make it idempotent.
+                    let unset_template_query: String = format!(
+                        "
+                        DO $$
+                        BEGIN
+                            IF EXISTS(
+                                SELECT 1
+                                FROM pg_catalog.pg_database
+                                WHERE datname = {}
+                            )
+                            THEN
+                            ALTER DATABASE {} is_template false;
```

Comment:
Can this ALTER ever fail? (I'm not familiar with template databases so this might be a silly question, but I've become wary of dependencies between objects)

Reply:
> I'm not familiar with template databases so this might be a silly question, but I've become wary of dependencies between objects

Yeah, same for me. I just checked that for non-template db it works without any problem. Will check the alter db code for non-obvious dependencies

Reply:
I had a look on the `dropdb()` code in Postgres and other reasons could be
- db is used by some active logical replication slots or subscription (we should likely deal with it after allowing logical replication)
- db is used by other users, open connections (this could be annoying, but will be retried and should eventually succeed, I'm thinking about just adding FORCE into drop db)

There are could be other issues, but I hope they will be more system-related, not logic-related